### PR TITLE
add nodejs client library

### LINKED ISSUE #333: Document connecting to RisingWave with JavaScript
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

We can have a new section "Develop" for app developers to perform common tasks (connect, create sources & mvs, insert data, output data. For each of the page we'll have examples for different languages. 

### Reference

https://questdb.io/docs/develop/insert-data/

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add the client libray for Node.js. 

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #333 

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/dev/nodejs-client-libraries.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+:::
+
+Connecting to RisingWave and running a query is normally done together. Therefore, we include a basic query in the code. Replace it with the query that you want to run.
+
+```js
+const { Pool } = require('pg')
+
+const credentials = {
+ user: 'root',
```
Comment (cloudcarver) [inline docs/dev/nodejs-client-libraries.md]: inconsistent indent
```suggestion
   user: 'root',
```

File: docs/dev/nodejs-client-libraries.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  port: 4566,
+}
+ 
+const start = async () => {
+    const pool = new Pool(credentials);
+    const res = await pool.query("SELECT 1+2"); /*A basic query to ensure the connection is successful. 
+    Replace it with your own query. */
+    console.log(res); //Print out the results.
+    await pool.end();
+      }
```
Comment (cloudcarver) [inline docs/dev/nodejs-client-libraries.md]: nit: code style
```suggestion
}
```

File: docs/dev/nodejs-client-libraries.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    datagen.rows.per.second='15',
+    datagen.split.num = '1'
+) ROW FORMAT JSON`;
+
+const start = async () => {
+    const pool = new Pool(credentials);
+    const res = await pool.query(createsource);
+    console.log(res);
+    await pool.end();
+      }
```
Comment (cloudcarver) [inline docs/dev/nodejs-client-libraries.md]: nit: code style
```suggestion
}
```

File: docs/dev/nodejs-client-libraries.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    SUM(distance) as total_distance,
+    SUM(duration) as total_duration
+    FROM walk`;
+
+const start = async () => {
+    const pool = new Pool(credentials);
+    const res = await pool.query(createmv);
+    console.log(res);
+    await pool.end();
+      }
```
Comment (cloudcarver) [inline docs/dev/nodejs-client-libraries.md]: nit: code style
```suggestion
}
```

File: docs/dev/nodejs-client-libraries.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    password: 'secret',
+    port: 4566,
+}
+
+const start = async () => {
+    const pool = new Pool(credentials);
+    const res = await pool.query("SELECT * from counter;");
+    console.log(res.rows); // Print out only the actual data.
+    await pool.end();
+      }
```
Comment (cloudcarver) [inline docs/dev/nodejs-client-libraries.md]: nit: code style
```suggestion
}
```

File: docs/dev/nodejs-client-libraries.md (hunk 1 above)
Comment (hengm3467) [reply docs/dev/nodejs-client-libraries.md]: Thanks! @mikechesterwang

**Review by cloudcarver — APPROVED** [review]
LGTM

**Review by neverchanje — APPROVED** [review]

**Pull request merged**